### ISSUE TITLE
Cannot use virtual modules without specify implementation library in libraries

### ISSUE THREAD

**Issue by hackwaly** [melange-re/melange#658]
dune version: 3.9.1
ocaml version: 4.14.1
melange version: both 1.0.0 58f3bcfc7dca5eea708de8e50e9a5e0ceda66b3c reproduced

A miminal reproducible demo is uploaded here: 

https://github.com/hackwaly/minimal-melange-virtual-library

```
➜  minimal-melange-virtual-library git:(master) dune build @melange
File "app/app.ml", line 1:            
Error: Compiler_specific not found, it means either the module does not exist or it is a namespace
```

Howerver this version works
https://github.com/hackwaly/minimal-melange-virtual-library/commit/a9ea6c641391d382ae770863a388bcb90238540a

**Comment by anmonteiro** [comment]
Is this the same issue? https://github.com/ocaml/dune/issues/7104

**Comment by hackwaly** [comment]
Yes, it is.

**Comment by anmonteiro** [comment]
The dune-side fix will be in 3.15: https://github.com/ocaml/dune/issues/7104

**Comment by hackwaly** [comment]
Still not work in dune 3.15
